### PR TITLE
Style-Guide conforming names for Linear-Solvers

### PR DESCRIPTION
**Pull request by philbucher**
With this PR I am updating the names of the linear-solvers in json 
e.g. `"solver_type" : "SuperLUSolver"` => `"solver_type" : "super_lu"`

Previously it was not consistent with the style-guide, now I unified everything

Also now it is consistent btw mpi and non-mpi

The tests are updated and running

Backwards-compatibility is ensured, by replacing the old names with the new names (and giving a deprecation-warning)

I am quite happy that I could finally unify this! This is a part of Kratos that I have been struggling with since the beginning: I never new what names the linear-solvers had.
Now this is quite intuitive and consistent with the rest of the code :D

### CODE REVIEW

File: applications/EigenSolversApplication/custom_python/add_custom_solvers_to_python.cpp
Comment (loumalouomega) [inline applications/EigenSolversApplication/custom_python/add_custom_solvers_to_python.cpp:116]: Are you sure you want to remove retrocompatibility?

**Review by loumalouomega — CHANGES_REQUESTED** [review]
I am not happy with breaking retrocompatibility, wait for @KratosMultiphysics/technical-committee

**Comment by philbucher** [comment]
I am not breaking retrocompatibility, read my message :)
> Backwards-compatibility is ensured, by replacing the old names with the new names (and giving a deprecation-warning)

**Review by loumalouomega — APPROVED** [review]
Oki

**Comment by philbucher** [comment]
Still this should be reviewed and approved by the committee

**Review by loumalouomega — APPROVED** [review]

**Comment by loumalouomega** [comment]
> Still this should be reviewed and approved by the committee

As you wish

**Comment by pooyan-dadvand** [comment]
As it changes the input (even with backward compatibility) I would suggest to be reviewed by the @KratosMultiphysics/interface-committee

**Comment by dbaumgaertner** [comment]
This indeed is a long necessary unification. Given that backward compatibility is ensured I would give it a go from @KratosMultiphysics/interface-committee side.

**Comment by philbucher** [comment]
If none objects then I would go ahead with merging, since then @rubenzorrilla and I can finish the cleanup/updates of the linear-solvers in the interface

**Comment by pooyan-dadvand** [comment]
@dbaumgaertner is this your opinion as a member of @KratosMultiphysics/interface-committee or is a consensus of all members?

**Comment by dbaumgaertner** [comment]
@pooyan-dadvand I am currently waiting for the response of the others and will let you know.

**Comment by dbaumgaertner** [comment]
Approving with consensus from @KratosMultiphysics/interface-committee

**Review by jcotela — APPROVED** [review]
The @KratosMultiphysics/technical-committee approves too. :+1:

**Comment by philbucher** [comment]
@jginternational @rubenzorrilla I will make a PR in the Interface to update the names

**Pull request merged**
